### DISCUSSION TITLE
Add auto-update configuration options

### DISCUSSION THREAD

**Discussion by vertigo220** [Squirrel/Squirrel.Windows#1843] [Ideas]
I don't like when software automatically downloads stuff without my permission in any circumstances, but especially given that I'm currently on a slow and somewhat metered connection, it's an especially big problem. I just saw that [Element](https://github.com/vector-im/element-web/) had an update available and debated whether I wanted to download another >100MB file after I just downloaded it to install it the other day. I decided to go ahead just so I could be sure to be using the latest version before submitting some feature requests and bug reports I have planned, and was surprised to find it updated instantly, meaning it had already downloaded it. This means it's using data I don't necessarily want used, downloading data to my computer without my knowledge or permission, and slowing down my other online activities due to using my limited bandwidth behind the scenes.

I [submitted an issue there](https://github.com/vector-im/element-web/issues/24253#issuecomment-1397113081), but was informed it's out of their hands, and it appears this is the repo where such a change needs to be requested and made in order for apps that use this for their updating to be able to allow the user to control the behavior. Since issues here appear to be for bugs only, I'm doing so here.

Ideally, there should be an option in the settings like with Windows updates that allows the user to choose how to handle updates:

- notify only
- download and notify
- download and install/restart automatically

**Comment by anaisbetts** [comment]
>  but was informed it's out of their hands,

This is incorrect. Update policy is 100% decided by the application for Squirrel applications

**Reply by vertigo220** [reply]
Thanks. Unfortunately, it seems everyone is saying they have no control over it and the issue lies with someone else. I [created an issue on electron's repo](https://github.com/electron/electron/issues/36958), so I guess we'll see what they say about it.

**Reply by anaisbetts** [reply]
The Electron auto-update API might have this restriction, because it is basically 100% modeled on Squirrel.Mac's API and Squirrel.Windows was added as an afterthought. You can always explicitly invoke update.exe as described in the documentation to get around this though, and indeed, Squirrel.Windows's check for update call was explicitly designed to be usable from node.js / Electron
